src/chain/chain.rs

```diff
@@ -304,6 +317,20 @@ impl Chain {
             | AccumulatingEvent::Relocate(_) => (),
         }
 
+        let start_churn_event = match event {
+            AccumulatingEvent::Online(_) | AccumulatingEvent::Offline(_) => true,
+            _ => false,
+        };
+
+        if start_churn_event {
+            if self.process_churn {
+                self.state.churn_event_backlog.push_front(event);
+                return Ok(None);
+            } else {
+                self.process_churn = true;
```

Comment:
One potential issue about this is that you expect the `Online` / `Offline` to always be followed by `SectionInfo`. But what about nodes that join but are not promoted to elders? Wouldn't we block any further churn events until someone else is promoted/demoted? This is not a problem in the current codebase, because we always promote everyone. But it won't stay that way for long.

Reply:
Thinking about this problem, maybe it's not an issue. We would always know if we want to promote a new node immediately or not. If we decide to not promote then, we can clear the `progress_churn` flag explicitly (and process the next backlogged event, if any).